transport/server.cc

```diff
@@ -237,24 +239,44 @@ cql_server::do_accepts(int which, bool keepalive, socket_address server_addr) {
             fd.set_keepalive(keepalive);
             auto conn = make_shared<connection>(*this, server_addr, std::move(fd), std::move(addr));
             ++_connects;
-            ++_connections;
-            // Move connection to the background, monitor for lifetime and errors.
-            (void)conn->process().then_wrapped([this, conn] (future<> f) {
+            (void)advertise_new_connection(conn).then([this, conn] {
+                ++_connections;
+                // Monitor connection's lifetime and errors in the background.
```

Comment:
The comment should not be moved, since it (tries to) explain the cast to void. In general casting a future to void is a no-no so we require an explanation about each call site.

I think the ++_connections should remain in its place. Consider a very slow advertise_new_connection(), we could have lots of new connections stuck there, with the connections metric not showing any change, because it is only updated after a_n_c().

Reply:
Here's my new take on this: https://github.com/jul-stas/scylla/blob/99d8441fd408c306fa09c20655487d5a5370eb8c/transport/server.cc#L240
I tried hybrid approach, with following assumptions:
- if `a_n_c()` fails exception is logged and connection proceeds normally
- if `conn->process()` fails, exception is logged and `unadvertise...()` is called anyway
- if both `conn->process()` and `unadvertise...()` fail then `nested_exception` is logged and accept() loops normally
It has its flaws (eg. unadvertise is called in the same fashion regardless of scenario) but it's the best I could think of ATM.